x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/session/Cursor.java

```diff
@@ -14,12 +14,35 @@
  * Information required to access the next page of response.
  */
 public interface Cursor extends NamedWriteable {
+
+    class Page {
+        private final RowSet rowSet;
+        private final Cursor next;
+
+        public Page(RowSet rowSet, Cursor next) {
+            this.rowSet = rowSet;
+            this.next = next;
+        }
+
+        public RowSet rowSet() {
+            return rowSet;
+        }
+
+        public Cursor next() {
+            return next;
+        }
+
+        public static Page end(RowSet rowSet) {
```

Comment:
What does "end" naming suppose to refer to? To me, initially, it was confusing as to what the functionality of it is, and the association with the name. Imo, it's not the end of a page, but the **last** page.